toolchain/sem_ir/typed_insts.h

```diff
@@ -402,11 +402,11 @@ struct ClassInit {
 
 struct ClassType {
   static constexpr auto Kind =
-      InstKind::ClassType.Define<Parse::AnyClassDeclId>("class_type");
+      InstKind::ClassType.Define<Parse::NodeId>("class_type");
```

Comment:
I'm not immediately seeing a change that requires this, so I assume this is a cleanup similar to the GenericClassType thing. Should types be using InvalidNodeId in general?

Reply:
We now create these in response to function calls, so we can also have a call expression here. (And in general we don't know that a call will be a syntactic call.)